From 9f061a617c2f2dd06b4ced94ac3ef75505dd02f5 Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Thu, 20 Oct 2022 12:00:11 +0300
Subject: [PATCH] [SW-108178] gaudi3: add pcie_lbw_rr_mstr_if configuration

When Efuse RR bit is set, HW FSM configures pcie_lbw_rr_mstr_if RR
registers along with pcie_elbi, pcie_mstr and nrtr. But this was missed
in ROM LBW configurations due to which host pcie cannot access PCI_CFG
LBW address space in ROM recovery mode when efuse RR bit is set.
(fyi, when Efuse RR bit is set, HW FSM blocks LBW and HBW in normal boot
flow and ROM unblocks the same in ROM recovery mode)

This patch fixes the above mentioned mismatch related to LBW RR
configurations.

Change-Id: I24ec94e68057d2746b7fc9d1a6a5d0ced08b5eb0
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 4dcdf82..196ecbd 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -179,6 +179,16 @@
 			rr_ctx->rr_idx);
 
 	regmap_writel(rr_ctx->data_min,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_MIN_0 + rr_dist);
+	regmap_writel(rr_ctx->data_max,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_MAX_0 + rr_dist);
+	regmap_writel(rr_ctx->en_val,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_EN_0 + rr_dist);
+
+	security_rr_update_hit_reg(mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_HIT_AW,
+			rr_ctx->rr_idx);
+
+	regmap_writel(rr_ctx->data_min,
 			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MIN_0 + rr_dist);
 	regmap_writel(rr_ctx->data_max,
 			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MAX_0 + rr_dist);
@@ -214,6 +224,17 @@
 			rr_ctx->rr_idx);
 
 	regmap_writel(rr_ctx->data_min,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_MIN_SHORT_0 + rr_dist);
+	regmap_writel(rr_ctx->data_max,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_MAX_SHORT_0 + rr_dist);
+	regmap_writel(rr_ctx->en_val,
+			mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_RANGE_EN_SHORT_0 + rr_dist);
+
+	security_rr_update_hit_reg(mmD0_PCIE_LBW_RR_MSTR_IF_RR_LBW_PRIV_HIT_AW,
+			rr_ctx->rr_idx);
+
+
+	regmap_writel(rr_ctx->data_min,
 			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MIN_SHORT_0 + rr_dist);
 	regmap_writel(rr_ctx->data_max,
 			mmD0_NRTR1_NW12_DEC1_CTRL_RR_LBW_PRIV_RANGE_MAX_SHORT_0 + rr_dist);
